src/rgw/driver/sfs/sfs_gc.h

```diff
@@ -67,14 +68,16 @@ class SFSGC : public DoutPrefixProvider {
   std::string get_cls_name() const { return "SFSGC"; }
 
  private:
-  void process_deleted_buckets();
+  // Return false if it was forced to exit because max process time was met
+  // which means there are still objects to be deleted
+  bool process_deleted_buckets();
+  bool process_deleted_objects();
+  bool delete_bucket(const std::string& bucket_id);
+  bool delete_pending_objects_data();
 
-  void delete_objects(const std::string& bucket_id);
-  void delete_versioned_objects(const Object& object);
+  bool process_time_elapsed() const;
 
-  void delete_bucket(const std::string& bucket_id);
-  void delete_object(const Object& object);
-  void delete_versioned_object(const Object& object);
+  sqlite::DBDeletedObjectItems pending_objects_to_delete;
```

Comment:
Suggest to rename this, as it contains object that are already deleted just not from the file system. From the point of view of the user, the objects are gone. From our point they are pending to delete from the filesystem at best or orphaned at worst.